lms/djangoapps/verify_student/services.py

```diff
@@ -56,3 +56,21 @@ def start_verification(self, course_id, related_assessment, item_id):
             )
         )
         return re_verification_link
+
+    def skip_verification(self, checkpoint_name, user_id, course_id):
+        """Create the add verification attempt
+
+        Args:
+            course_id(str): A string of course_id
+            user_id(str): User Id string
+            checkpoint_name(str): Verification checkpoint name
+
+        Returns:
+            None
+        """
+        course_key = CourseKey.from_string(course_id)
+        checkpoint = VerificationCheckpoint.objects.get(course_id=course_key, checkpoint_name=checkpoint_name)
+
+        # if user do not already skipped the attempt for this course only then he can skip
+        if not SkippedReverification.check_user_skipped_reverification_exists(user_id, course_key):
+            SkippedReverification.add_skipped_reverification_attempt(checkpoint, user_id, course_key)
```

Comment:
I'm concerned about a race condition here.  It's not likely, but if this method is called multiple times for the same (user, course_key) combination, you might get an `IntegrityError` (when the uniqueness constraint is violated).

I think we should catch the integrity error (maybe logging a warning) without re-raising the exception, since at that point we know that the record exists.